src/ncp/ncp_base.cpp

```diff
@@ -2975,10 +2975,41 @@ ThreadError NcpBase::GetPropertyHandler_THREAD_RLOC16_DEBUG_PASSTHRU(uint8_t hea
 
 ThreadError NcpBase::GetPropertyHandler_THREAD_LOCAL_ROUTES(uint8_t header, spinel_prop_key_t key)
 {
-    // TODO: Implement get external route table
-    (void)key;
+    ThreadError errorCode = kThreadError_None;
+    otExternalRouteConfig external_route_config;
+    otNetworkDataIterator iter = OT_NETWORK_DATA_ITERATOR_INIT;
+    uint8_t flags;
 
-    return SendLastStatus(header, SPINEL_STATUS_UNIMPLEMENTED);
+    mDisableStreamWrite = true;
+
+    SuccessOrExit(errorCode = OutboundFrameBegin());
+    SuccessOrExit(errorCode = OutboundFrameFeedPacked(SPINEL_DATATYPE_COMMAND_PROP_S, header, SPINEL_CMD_PROP_VALUE_IS,
+                                                      key));
+
+    while (otNetDataGetNextRoute(mInstance, false, &iter, &external_route_config) == kThreadError_None)
```

Comment:
For me, the intention for how to use this property is to get the external routes in the partition's network data  so that we can ensure they are applied on the host side (e.g., by wpantund on a linux host for `wpan0` interface). The property name `LOCAL_ROUTES` is indeed confusing.

My suggestion is to change `LOCAL_ROUTES` to gives the external routrs from local network data and define a new spinel property for the partition's network datat external route entries...